### ISSUE TITLE
Dirty daemon shutdown

### ISSUE THREAD

**Issue by da-kami** [itchysats/itchysats#477]
We have reported this problem in the past, we still see errors like this when shutting down:

```
2021-11-04 18:20:53  INFO Received shutdown request. Waiting for pending I/O...
2021-11-04 18:20:55  WARN Wallet feed receiver not available, stopping wallet sync
2021-11-04 18:20:58 ERROR Closing connection to taker 907107d8-6ccb-4cce-9366-a477c9fd98da
2021-11-04 18:20:59 ERROR Server failed to shutdown cooperatively. Terminating.
```

On that particular shutdown the database was not properly merged, i.e. the `maker.sqlite-shm` and  `maker.sqlite-wal` file were present after terminating the application. 

Can we somehow ensure that the database always gets closed properly as part of the rocket shutdown? Or manage database outside of rocket, but ensure proper closing upon shutdown?

**Comment by da-kami** [comment]
@klochowicz I did not remember that I opened a ticket for this a while ago. Thought I ping you on it, because you mentioned that you are planning to tackle this. Would be great if we can close this once the work is done :)

**Comment by klochowicz** [comment]
#589 is a step towards achieving a clean shutdown - not the final step perhaps, but it's a step towards that goal :)

**Comment by thomaseizinger** [comment]
We might want to do a PoC with `warp` or a different HTTP framework to see if that helps.

**Comment by klochowicz** [comment]
I don't think it's Rocket's fault - we have already moved the db to not be managed by Rocket. Something's holding an open connection and prevents the `db.close()` from returning - it awaits until all the connections are closed.

It's easily verifiable - if we comment that one, rocket exits cleanly.

We can always extend the grace period in the shutdown to cover for that if cleanup takes a while
 (I tried with 20s, and the `db.close()` was still waiting to close all connections).
